Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -645,7 +652,21 @@ IPCReply NetKDRequestDevice::HandleNWC24CheckMailNow(const IOCtlRequest& request
 
 IPCReply NetKDRequestDevice::HandleNWC24DownloadNowEx(const IOCtlRequest& request)
 {
-  m_dl_list.ReadDlList();
+  if (m_dl_list.IsDisabled())
+  {
+    // Signal that the DL List is broken.
+    LogError(ErrorType::KD_Download, NWC24::WC24_ERR_BROKEN);
+    WriteReturnValue(NWC24::WC24_ERR_BROKEN, request.buffer_out);
+    return IPCReply(IPC_SUCCESS);
+  }
+
+  if (!m_dl_list.ReadDlList())
+  {
+    LogError(ErrorType::KD_Download, NWC24::WC24_ERR_BROKEN);
+    WriteReturnValue(NWC24::WC24_ERR_BROKEN, request.buffer_out);
+    return IPCReply(IPC_SUCCESS);
+  }
```

Comment:
Except for the comment these two blocks are identical. Unless you expect that to change in future PRs I would merge them.